src/crypto/crypto_context.cc

```diff
@@ -230,9 +460,21 @@ X509_STORE* NewRootCertStore() {
 
   if (!root_certs_vector_loaded) {
     if (per_process::cli_options->ssl_openssl_cert_store == false) {
+      std::vector<std::string> combined_root_certs;
+
       for (size_t i = 0; i < arraysize(root_certs); i++) {
+        combined_root_certs.emplace_back(root_certs[i]);
+      }
+
+      if (per_process::cli_options->use_system_ca) {
```

Comment:
This question predates this PR I guess, but shouldn't these options be per-env instead of per-process? @addaleax @jasnell

Reply:
FWIW existing code in the same area uses per_process for the same thing from my understanding:
https://github.com/nodejs/node/blob/1238f0afba8d44fb9155deeefbeb6bb71b62d632/src/crypto/crypto_context.cc#L242

(I don't know the difference as-of yet though)

Reply:
Yeah, ideally these should all be per-env configuration options I'd say 👍

Reply:
Is this required?

I gave it a go and `env` is not available in this function and it has a few callers, I'm not sure the impact of going through and adding args/ env all over the place and if thats the best approach.

A bit of guidance would be appreciated on this one.

Reply:
I think for this PR it is not required, having per-environment setting is a nice-to-have, but the unnecessarily global scope for certificates already predates this PR.